usecases/backup/scheduler.go

```diff
@@ -104,35 +104,42 @@ func (s *Scheduler) Restore(ctx context.Context, pr *models.Principal,
 
 func (s *Scheduler) BackupStatus(ctx context.Context, principal *models.Principal,
 	backend, backupID string,
-) (*models.BackupCreateStatusResponse, error) {
+) (*Status, error) {
 	path := fmt.Sprintf("backups/%s/%s", backend, backupID)
 	if err := s.authorizer.Authorize(principal, "get", path); err != nil {
 		return nil, err
 	}
-
 	store, err := coordBackend(s.backends, backend, backupID)
 	if err != nil {
 		err = fmt.Errorf("no backup provider %q, did you enable the right module?", backend)
 		return nil, backup.NewErrUnprocessable(err)
 	}
 
-	st, err := s.backupper.OnStatus(ctx, store, &StatusRequest{OpCreate, backupID, backend})
+	req := &StatusRequest{OpCreate, backupID, backend}
+	st, err := s.backupper.OnStatus(ctx, store, req)
 	if err != nil {
 		return nil, backup.NewErrNotFound(err)
 	}
-	// check if backup is still active
-	status := string(st.Status)
-	return &models.BackupCreateStatusResponse{
-		ID:      backupID,
-		Path:    st.Path,
-		Status:  &status,
-		Backend: backend,
-	}, nil
+	return st, nil
 }
 
-func (m *Scheduler) RestorationStatus(ctx context.Context, principal *models.Principal, backend, ID string,
-) (_ RestoreStatus, err error) {
-	return RestoreStatus{}, backup.NewErrUnprocessable(fmt.Errorf("not implemented"))
+func (s *Scheduler) RestorationStatus(ctx context.Context, principal *models.Principal, backend, backupID string,
+) (_ *Status, err error) {
+	path := fmt.Sprintf("backups/%s/%s/restore", backend, backupID)
+	if err := s.authorizer.Authorize(principal, "get", path); err != nil {
+		return nil, err
+	}
+	store, err := coordBackend(s.backends, backend, backupID)
+	if err != nil {
+		err = fmt.Errorf("no backup provider %q, did you enable the right module?", backend)
```

Comment:
The original `err` is being swallowed here. Would it be better to include the original `err` in this reassignment?

Reply:
Done